### PR TITLE
chore: prepare for 7.11.x release branch

### LINKED ISSUE #543: Setup branch for 7.11.x e2e-testing repo and retire prior branch from Jenkins nightly tests
Every release we should:
retire the current GA product test line (7.10.x) as the new one goes out
branch the tests for the new coming GA (7.11.x)
 - this needs to be manually done in repo, with some variable / inputs in Jenkinsfiles updated
 - for Fleet we may want to update the 'stale' version Agent in test in master and 7.x line?
 - any other updates, like Helm Chart versions etc?  we should gather all info in one ticket so it is easier to repeat.  :)

All the while:
- .x branch continues on.  It becomes the initial tests for next GA (this means 7.12 now)
- master (8.0) continues on.  

Note, we missed shutting down 7.9 last time, but its ok - we can do it now, too.

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It adds 7.11.x to backport config, also setting a regex to monitor the valid branches in JJBB.

## Why is it important?
We want to retire 7.9.x and enable 7.11.x

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [x] No other descriptor containing branches is affected

## Related issues

- Closes #543

## Follows-up
No need to backport this to maintenance branches. Once merged, we could create 7.11.x branch and update references to product versions

### CODE REVIEW

File: .ci/jobs/e2e-testing-integrations-daily-mbp.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@
     scm:
       - github:
           branch-discovery: no-pr
-          head-filter-regex: '(master|\d+\.\d+\.x|\d+\.x)'
+          head-filter-regex: '(master|7\.x|7\.11\.x|7\.10\.x)'
```
Comment (mdelapenya) [inline .ci/jobs/e2e-testing-integrations-daily-mbp.yml:12]: Would this be better/simpler/more readable?
```suggestion
          head-filter-regex: '(master|7\.x|7\.1[01]\.x)'
```

**Review by cachedout — APPROVED** [review]

File: .ci/jobs/e2e-testing-integrations-daily-mbp.yml (hunk 1 above)
Comment (v1v) [reply .ci/jobs/e2e-testing-integrations-daily-mbp.yml:12]: I'd say explicit regex might be easier to read in the future

**Review by v1v — APPROVED** [review]

File: .ci/jobs/e2e-testing-integrations-daily-mbp.yml (hunk 1 above)
Comment (mdelapenya) [reply .ci/jobs/e2e-testing-integrations-daily-mbp.yml:12]: You mean the first one, or the condensed, second one?

Comment (v1v) [reply .ci/jobs/e2e-testing-integrations-daily-mbp.yml:12]: I prefer the first one, no the suggested change

Comment (mdelapenya) [reply .ci/jobs/e2e-testing-integrations-daily-mbp.yml:12]: Thanks for the feedback! Merging now

File: .ci/jobs/e2e-testing-mbp.yml [hunk 2]
```diff
@@ -9,6 +9,7 @@
     scm:
       - github:
           branch-discovery: no-pr
+          head-filter-regex: '(master|7\.x|7\.11\.x|7\.10\.x)'
```
Comment (mdelapenya) [inline .ci/jobs/e2e-testing-mbp.yml:12]: @elastic/observablt-robots could it be possible that we skipped monitoring PRs with this change?

Comment (kuisathaverat) [reply .ci/jobs/e2e-testing-mbp.yml:12]: yep, you'd need to add `PR-.*` see https://github.com/elastic/apm-integration-testing/blob/master/.ci/jobs/apm-integration-tests.yml#L14

Comment (mdelapenya) [reply .ci/jobs/e2e-testing-mbp.yml:12]: And tags?

Comment (mdelapenya) [reply .ci/jobs/e2e-testing-mbp.yml:12]: Sent a fix here: https://github.com/elastic/e2e-testing/pull/571. 🙏  👀

**Pull request merged**